hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -277,6 +280,25 @@ private void assertThrottlingPreconditions(
                 throw new PreCheckException(NOT_SUPPORTED);
             }
         }
+
+        long gas =
+                switch (txInfo.functionality()) {
+                    case CONTRACT_CREATE -> txInfo.txBody()
+                            .contractCreateInstanceOrThrow()
+                            .gas();
+                    case CONTRACT_CALL -> txInfo.txBody().contractCallOrThrow().gas();
+                    case ETHEREUM_TRANSACTION -> Optional.of(txInfo.txBody()
+                                    .ethereumTransactionOrThrow()
+                                    .ethereumData()
+                                    .toByteArray())
+                            .map(EthTxData::populateEthTxData)
+                            .map(EthTxData::gasLimit)
+                            .orElse(0L);
+                    default -> 0L;
+                };
+        if (gas < 0) {
```

Comment:
Again, my question would be: Why `<0` instead of `<=0` or even `<= 21000`.   

(Don't know the answer here or above, which is why I'm asking.)

In this case you'd have to signal the `default` case - i.e., not a contract-related operation throttled by gas - separately than using `0L`.

Reply:
Needs a test in `IngestCheckerTest.ThrottleTests`?

Actually, I don't see a test for any of the conditions in `assertThrottlingPreconditions`.  Also actually, `assertThrottlingPreconditions` seems misnamed?  Until this code you just added it doesn't seem to be checking _throttling_ preconditions at all, just some interesting system preconditions.

Reply:
Most of the preconditions are checked during `pureChecks` methods on each transaction type. That's where we check whether the gas limit is <=intrinsicGas. But, checking throttles comes earlier in the ingest workflow than running pureChecks. Before we check throttles, we do just the bare minimum checks necessary to ensure that the throttle checking code will succeed. That's what this method is about.

Reply:
And yes, I'll add that test too.

Reply:
Test added